### ISSUE TITLE
Extend the Login IP Filters functionality to regular users

### ISSUE THREAD

**Issue by davidjrh** [dnnsoftware/Dnn.Platform#5373]
## Description of problem
The current code base that implements the logic for the Login IP Filters only takes in consideration Superusers and users under the Administrator role. While this is not mentioned anywhere in the persona bar settings, I thought it was intended for all users. 

The code below is on https://github.com/dnnsoftware/Dnn.Platform/blob/develop/DNN%20Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs#L1298

```
  // Obtain the current client IP
  var userRequestIpAddressController = UserRequestIPAddressController.Instance;
  var ipAddress = userRequestIpAddressController.GetUserRequestIPAddress(new HttpRequestWrapper(this.Request));

  // check if the user is an admin/host and validate their IP
  if (Host.EnableIPChecking)
  {
      bool isAdminUser = objUser.IsSuperUser || objUser.IsInRole(this.PortalSettings.AdministratorRoleName);
      if (isAdminUser)
      {
          var clientIp = ipAddress;
          if (IPFilterController.Instance.IsIPBanned(clientIp))
          {
              PortalSecurity.Instance.SignOut();
              this.AddModuleMessage("IPAddressBanned", ModuleMessage.ModuleMessageType.RedError, true);
              okToShowPanel = false;
              break;
          }
      }
  }
```

I understand that I can restrict the access by IP using other mechanisms such as touching the web.config, adding rules at hosting level, etc. but extending this to regular users would allow Authenticatin Extensions like the Azure AD module to use this configuration for several purposes such as the one mentioned at https://github.com/davidjrh/dnn.azureadprovider/issues/50

Is there any reason to restrict this to superusers and admins? Could be extended to regular users? If we agree, I'm happy to do a PR.

**Comment by valadas** [comment]
I am not disagreeing but if that is done, it needs to be an option to not break how it works now. A lot of people do it to prevent getting hacked as a bad admin/host only.

**Comment by mitchelsellers** [comment]
I personally had not noticed this limitation. I think adding a option for who it applies to would make sense here.

**Comment by jeremy-farrance** [comment]
I like this. Key points:
- the existing UI needs its labelling and info improved ASAP (should that be a separate Issue and tackled soon?)
- if enhanced, the default/existing behavior should not change

_My off the cuff opinion (possibly less than 2 cents) or suggestion for consideration:_ 

If this is enhanced, the new behavior should be tied to Roles only, no option for individual users. For example, an Allow rule for  "All Logins" would effectively be set on the "Registered Users" role.

**Comment by davidjrh** [comment]
Cool, based on the comments above the following UI would allow to extend it to "Registered Users" and be backwards compatible. Note that the role would need to be restricted to System roles because this would be a Global Setting. 

![image](https://user-images.githubusercontent.com/489976/200120934-8f4c890e-62b7-4a1f-882d-33500e438250.png)

Agree?

**Comment by sleupold** [comment]
but roles "registered Users" and "administrators" are  system defined, but individual per portal (and you shouldn't rely on the role name)

**Comment by davidjrh** [comment]
Of course, the idea is to populate that dropdown with the current portal roles, and change the currenct check `objUser.IsInRole(this.PortalSettings.AdministratorRoleName)` for something like 'objUser.IsInRole(this.PortalSettings.IPAddressCheckRoleName)`
